src/outputs.ts

```diff
@@ -0,0 +1,61 @@
+import { Outpoint, Output, PrivateKey, RecipientAddress } from './interface.ts';
+import { calculateSumOfPrivateKeys, hashOutpoints } from './utility.ts';
+import { decodeSilentPaymentAddress, serialiseUint32 } from './encoding.ts';
+import { SHA256, secp256k1 } from 'bcrypto';
+
+export const createOutputs = (
+    inputPrivateKeys: PrivateKey[],
+    outpoints: Outpoint[],
+    recipientAddresses: RecipientAddress[],
+    hrp: string = 'tsp',
+): Output[] => {
+    const sumOfPrivateKeys = calculateSumOfPrivateKeys(inputPrivateKeys);
+    const outpointHash = hashOutpoints(outpoints);
+
+    const paymentGroups = new Map<
+        string,
+        { spendKey: Buffer; amount: number }[]
+    >();
+
+    for (const { address, amount } of recipientAddresses) {
```

Comment:
Cant we use `recipientAddresses.forEach()` to do this parallely or do we need this to happen sequentially?

Reply:
Just convention